operations/automated-upgrades/flymd.md

```diff
@@ -0,0 +1,148 @@
+
+```pre
+$ pip install -r requirements.txt
+```
+Note, if your Python environment has not been configured to live in your home directory you may have to:
+
+1. prefix the above command with 'sudo' to install in system directories
+1. or, preferably, this is probabaly a good time to get familar with the [anyenv](https://github.com/riywo/anyenv) and [pyenv](https://github.com/pyenv/pyenv) tools
+
+The environment also relies on a handful of Vagrant plugins. Assuming you've already installed Vagrant, you can install the plugins like so:
+
+```pre
+$ vagrant plugin install vagrant-hosts
+$ vagrant plugin install vagrant-auto_network
+```
+
+#### Operating the demo environment
+
+Vagrant/Virtualbox can take a few minutes to spin the VMs (serial operation :\) plus there's a minimal amount of work to be done installing supporting packages so it makes sense to get that out of the way first.
+
+```pre
+$ vagrant up /consul/
+```
+
+Next, let's provision the Consul cluster composed of the "a" nodes.
+
+```pre
+$ vagrant up /consula/ --provision-with configure,validate
+```
+
+You should see green [Chef Inspec](https://www.chef.io/inspec/) output for each VM indicating that pre-flight checks have executed with success. If you see red or orange checks that means something is amiss. The best recourse is to run the the similar command shown below. The environment has been designed to be as idempotent as possible despite not using an idempotent configuration management tool. If the problems persist then feel free reach out using the contact information included with this doc.
+
+It might also be useful to note that you can re-rerun *just the validation* tests at any time like so:
+
+```pre
+$ for i in consula0 consula1 consula2; do vagrant provision $i --provision-with configure,validate; done
+```
+
+It might be useful to note that you can re-run just the validation tests for any provisioned and running node at any time. The example below is for just 'consula0':
+
+```pre
+$ vagrant provision consula0 --provision-with validate
+```
+
+Assuming that all of the nodes a running and that the "a" nodes are already configured as a Consul cluster, you should now be able to see the Consul cluster UI on ```http://localhost:8500```.
+
+To start the upgrade:
+
+```pre
+$ doitlive play vms/consul/consul-upgrade.sh
+```
+
+[doitlive](https://doitlive.readthedocs.io/en/latest/) is a tool for scripting command-line demos. Once doitlive has been invoked you can type random keys and it will send the scripted input to stdin of your shell. Ctrl-C will break out of the scripted demo.
+
+Note that the ```vms/consul/consul-upgrade.sh``` script resets the cluster by removing the directory ```/var/hashicorp/consul``` and restarting the Consul processes. 
+
+That's it. Learn. Play. Send bug reports and questions. Enjoy.
+
+## Contact
+
+* Nathan Valentine - nathan @ hashicorp com | nrvale00 @ gmail com
+* https://github.com/hashicorp-guides
```

Comment:
I'd remove this reference

Reply:
That file was an accidental check-in of a temporary file. I thought I had already purged it but will do so.